scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,836 @@
+                recMan.close();
+                recMan = null;
+            } catch (IOException e) {
+                // ignored
+            }
+        }
+        return true;
+    }
+
+    private <T extends Serializable> Channel<T> getChannel(String name) throws InvalidChannelException {
+        if (inMemoryChannels.containsKey(name)) {
+            return inMemoryChannels.get(name);
+        } else if (persistedChannels.containsKey(name)) {
+            return persistedChannels.get(name);
+        }
+        throw new InvalidChannelException("Channel " + name + " does not exist");
+    }
+
+    @Override
+    public boolean createChannel(String originator, TaskId taskid, String name, boolean isPersistent)
+            throws IOException {
+        try {
+            boolean alreadyExistingChannel = deleteChannel(originator, taskid, name);
+            Channel newChannel = new Channel();
+            if (isPersistent) {
+                persistedChannels.put(name, newChannel);
+                logWithContext(taskid, null, "Created new persistent channel " + QUOTE + name + QUOTE, Level.INFO);
+                recMan.commit();
+            } else {
+                logWithContext(taskid, null, "Created new memory channel " + QUOTE + name + QUOTE, Level.INFO);
+                inMemoryChannels.put(name, newChannel);
+            }
+            return !alreadyExistingChannel;
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public boolean deleteChannel(String originator, TaskId taskid, String name) throws IOException {
+        try {
+            if (persistedChannels.containsKey(name)) {
+                persistedChannels.remove(name);
+                logWithContext(taskid, null, "Deleted persistent channel " + QUOTE + name + QUOTE, Level.INFO);
+                recMan.commit();
+                return true;
+            } else if (inMemoryChannels.containsKey(name)) {
+                inMemoryChannels.remove(name);
+                logWithContext(taskid, null, "Deleted memory channel " + QUOTE + name + QUOTE, Level.INFO);
+                return true;
+            } else {
+                return false;
+            }
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public boolean createChannelIfAbsent(String originator, TaskId taskid, String name, boolean isPersistent)
+            throws IOException {
+        if (inMemoryChannels.containsKey(name) || persistedChannels.containsKey(name)) {
+            return false;
+        } else {
+            createChannel(originator, taskid, name, isPersistent);
+            return true;
+        }
+    }
+
+    private void commitIfNeeded(String channel) throws IOException {
+        if (persistedChannels.containsKey(channel)) {
+            persistedChannels.put(channel, persistedChannels.get(channel));
```

Comment:
I am puzzled by this line. Could you say what it does?

Reply:
Oh yes, I understand, I should have put a comment. Persistence is triggered on the record manager only when you do put operations. When you do:
persistedChannels.get(channel).dosomethingOnTheObject()
recMan.commit()
Nothing is commited.

Reply:
Make sense. Thank you